tests/tensorizer_loader/test_tensorizer.py

```diff
@@ -256,6 +281,65 @@ def test_tensorizer_with_tp(vllm_runner):
             tensor_parallel_size=2,
         )
 
+@pytest.mark.skip("Requires multiple GPUs")
+def test_deserialized_encrypted_vllm_model_with_tp_has_same_outputs(vllm_runner,
+                                                                    tmp_path):
+    model_ref = "EleutherAI/pythia-1.4b"
+    # record outputs from un-sharded un-tensorized model
+    base_model = vllm_runner(
+        model_ref,
+    )
+    outputs = base_model.generate(prompts, sampling_params)
+
+    base_model.model.llm_engine.model_executor.shutdown()
+    del base_model
+    cleanup()
+
+    # load model with two shards and serialize with encryption
+    model_path = str(tmp_path / (model_ref + "-%02d.tensors"))
+    key_path = tmp_path / (model_ref + ".key")
+
+    config_for_serializing = TensorizerConfig(
+        tensorizer_uri=model_path,
+        encryption_keyfile=key_path,
+    )
+    # FIXME: launching multiple multiprocessing servers within the same program
+    # results in a hang... launch serialization in a separate process as a work
+    # around
```

Comment:
Will assume this is a WIP

Reply:
I think the functionality supporting sharding is still useful even if the tests can't all pass together... Hopefully we can find a fix to this issue though. I'm planning to do a bit more digging into it to see if I can find anything, but let me know if you have ideas!

Reply:
Found a work-around! The hang seems to be caused by the custom all reduce kernel not being completely cleaned up. The tests no longer hang if I use `disable_custom_all_reduce=True`. With that change I have removed this FIXME note and changed to skipif for the tests.